File: 4.AHIFS/BSD[POS1]/Programme/004Judth_Blaschke_SessionRaterV2/SessionRaterClient/Session.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SessionRaterClient
{
    public class Session
    {
        public int Id { get; }
        public string Title { get; set; }
        public string Speaker { get; set; }
        public SessionState CurrentSessionState { get; set; }
        public HashSet<Rating> Ratings { get; set; }

        public Session(int id, string Title, string Speaker)
        {
            this.Id = id;
            this.Title = Title;
            this.Speaker = Speaker;
            this.Ratings = new HashSet<Rating>();

            CurrentSessionState = SessionState.Created;
        }


        #region public-Methods
        public void addRating(Rating newRating)
        {
            CurrentSessionState = SessionState.Evaluated;
            Ratings.Add(newRating);
        }

        public void closeSession()
        {
            CurrentSessionState = SessionState.Closed;
        }
        public override string ToString()
        {
            return "Title: " + Title + "    Speaker: " + Speaker + "    Status: " + CurrentSessionState.ToString();
        }
        #endregion
    }
}

File: 4.AHIFS/BSD[POS1]/Programme/004Judth_Blaschke_SessionRaterV2/SessionRaterJS/Session.js
var enumSessionState = require('./SessionSate.js');
function Session(id, title, speaker){
    //properties
    var id = id;
    var title = title;
    var speaker = speaker;
    var ratings = [];
    var sessionState = enumSessionState.Created;

    //Methods
    this.addRating = _addRating;
    this.closeSession = _closeSession;
    this.toString = _toString;
    this.getTitle = _getTitle;
    this.getID = _getID;
    this.getSpeaker = _getSpeaker;
    this.getSessionState = _getSessionState;
    
    

    function _addRating(newRating){
        ratings.push(newRating);
        sessionState = enumSessionState.Evaluated;
    }
    
    function _closeSession(){
        sessionState = enumSessionState.Closed;
    }
    
    function _getID(){
        return id;
    }
     
    function _getTitle(){
        return title;
    }
    
    function _getSpeaker(){
        return speaker;
    }
    
    function _getSessionState(){
        return sessionState;
    }

    function _toString(){
        var ratingString = "{";
        ratings.forEach(function (e){ ratingString += " " + e.toString() + ";" });
        ratingString += "}";
        return "ID: " + id + ", Title " + title + ", Speaker: " + speaker + ", Number of ratings: " + ratingString + " State: " + sessionState + ";";
    }

}

module.exports = Session;